core/src/main/java/org/apache/struts2/components/Debug.java

```diff
@@ -50,31 +49,46 @@ public Debug(ValueStack stack, HttpServletRequest request, HttpServletResponse r
     public void setReflectionProvider(ReflectionProvider prov) {
         this.reflectionProvider = prov;
     }
-    
+
     protected String getDefaultTemplate() {
         return TEMPLATE;
     }
 
     public boolean start(Writer writer) {
         boolean result = super.start(writer);
 
-        ValueStack stack = getStack();
-        Iterator iter = stack.getRoot().iterator();
-        List stackValues = new ArrayList(stack.getRoot().size());
-        while (iter.hasNext()) {
-            Object o = iter.next();
-            Map values;
-            try {
-                values = reflectionProvider.getBeanMap(o);
-            } catch (Exception e) {
-                throw new StrutsException("Caught an exception while getting the property values of " + o, e);
+        if (showDebug()) {
+            ValueStack stack = getStack();
+            Iterator iter = stack.getRoot().iterator();
+            List stackValues = new ArrayList(stack.getRoot().size());
+            while (iter.hasNext()) {
+                Object o = iter.next();
+                Map values;
+                try {
+                    values = reflectionProvider.getBeanMap(o);
+                } catch (Exception e) {
+                    throw new StrutsException("Caught an exception while getting the property values of " + o, e);
+                }
+                stackValues.add(new DebugMapEntry(o.getClass().getName(), values));
             }
-            stackValues.add(new DebugMapEntry(o.getClass().getName(), values));
+
+            addParameter("stackValues", stackValues);
         }
+        return result;
+    }
 
-        addParameter("stackValues", stackValues);
+    @Override
+    public boolean end(Writer writer, String body) {
+        if (showDebug()) {
+            return super.end(writer, body);
+        } else {
+            popComponentStack();
+            return false;
+        }
+    }
 
-        return result;
+    protected boolean showDebug() {
+        return (devMode || "false".equalsIgnoreCase(disabled));
```

Comment:
It's exactly the same, there is no another `disabled` property in `Debug` tag.

Reply:
> $(".selector").attr("disabled", "false")

I meant your mentioned code above is at client side html and is not equal to that `disabled` java property at server side which this PR uses.

Reply:
Yes but that property is settable via client -> `<input disabled="false"/>` or `?disabled=true` - I'm not saying it's true in this case, that's why I asked for second pair of  👀

Reply:
Are you sure?! I'm sure that server side java property (disabled) is not settable via client. Client cannot execute `<s:debug disabled="true"/>` as this is a server side tag not html tag. I checked output of `<s:debug...`; it generates only read-only not postable html elements.

Reply:
That's why asked for second pair of 👀

Reply:
To be clear: `disabled` is a user settable property, that's the purpose of this property, a pure client side thing. Right now it was used to control server side logic and I wonder if this is a good idea.

Reply:
Aha... yes I agree. However, `debug.ftl` does not use it currently.

@HedjuHor , I found all references of `UIBean.disabled` and saw it never used in java codes and always has been used inside `ftl` (free marker template) files. As Łukasz correctly warned, if you look it's usages inside `ftl` files, you'll see currently it's purpose is only to generate or not generate html `disabled` attribute. So I also think it may not a good idea to use it here, in not it's business ;)

@HedjuHor , @lukaszlenart , what do you feel using it inside `debug.ftl` instead as below:

```diff
Index: core/src/main/resources/template/simple/debug.ftl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- core/src/main/resources/template/simple/debug.ftl	(date 1515941681000)
+++ core/src/main/resources/template/simple/debug.ftl	(revision )
@@ -18,6 +18,8 @@
  * under the License.
  */
 -->
+<#if parameters.disabled!false>
+<#else >
 <script type="text/javascript">
 <!--
     function toggleDebug(debugId) {
@@ -86,3 +88,4 @@
     </#list>
 </table>
 </div>
+</#if>
\ No newline at end of file
Index: core/src/main/java/org/apache/struts2/components/Debug.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- core/src/main/java/org/apache/struts2/components/Debug.java	(date 1515941681000)
+++ core/src/main/java/org/apache/struts2/components/Debug.java	(revision )
@@ -41,6 +41,7 @@
     protected ReflectionProvider reflectionProvider;
 
 
+
     public Debug(ValueStack stack, HttpServletRequest request, HttpServletResponse response) {
         super(stack, request, response);
     }
@@ -57,38 +58,31 @@
     public boolean start(Writer writer) {
         boolean result = super.start(writer);
 
-        if (showDebug()) {
-            ValueStack stack = getStack();
-            Iterator iter = stack.getRoot().iterator();
-            List stackValues = new ArrayList(stack.getRoot().size());
-            while (iter.hasNext()) {
-                Object o = iter.next();
-                Map values;
-                try {
-                    values = reflectionProvider.getBeanMap(o);
-                } catch (Exception e) {
-                    throw new StrutsException("Caught an exception while getting the property values of " + o, e);
-                }
-                stackValues.add(new DebugMapEntry(o.getClass().getName(), values));
-            }
+        ValueStack stack = getStack();
+        Iterator iter = stack.getRoot().iterator();
+        List stackValues = new ArrayList(stack.getRoot().size());
+        while (iter.hasNext()) {
+            Object o = iter.next();
+            Map values;
+            try {
+                values = reflectionProvider.getBeanMap(o);
+            } catch (Exception e) {
+                throw new StrutsException("Caught an exception while getting the property values of " + o, e);
+            }
+            stackValues.add(new DebugMapEntry(o.getClass().getName(), values));
+        }
 
-            addParameter("stackValues", stackValues);
-        }
+        addParameter("stackValues", stackValues);
+
         return result;
     }
 
     @Override
     public boolean end(Writer writer, String body) {
-        if (showDebug()) {
-            return super.end(writer, body);
-        } else {
-            popComponentStack();
-            return false;
-        }
-    }
-
-    protected boolean showDebug() {
-        return (devMode || "false".equalsIgnoreCase(disabled));
+        if (!(devMode || "false".equalsIgnoreCase(disabled))) {
+            disabled = "true";
+        }
+        return super.end(writer, body);
     }
 
     private static class DebugMapEntry implements Map.Entry {
```

I have been tested above patch on this PR and passes all tests :)

Reply:
firstly, thank you to all
@yasserzamani 
- debug.ftl change looks nice.
- Debug.java i would keep my version with the 'showDebug' methode for performance reason. If developer keep the DebugTag in the Jsp-files and defMode is disabled it always unnecessarily fetch the stackValues

if